### PR TITLE
TLS Syslog tests

### PR DESCRIPTION
**Pull request by voidlock**
A big set of changes that includes

- refactoring to rebar3
- refactoring to a single CT Suite capable of testing both Plain TCP and TLS Syslog drain types
- adds a syslog drain endpoint using [ranch](https://github.com/ninenines/ranch)

### CODE REVIEW

**Comment by voidlock** [comment]
Would appreciate some feedback from @heroku/telemetry

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM voidlock/erlang:R16B03-1-onbuild
+FROM voidlock/erlang:17.5-onbuild
```
Comment (srid) [inline Dockerfile]: Any reason this is not using an official or public (and thus impersonal) docker image?

File: rebar.config [hunk 2]
```diff
@@ -1,24 +1,71 @@
...
- ,{ex_uri, "", {git, "git@git.herokai.com:ex_uri.git", "master"}}
- ,{heroku_crashdumps, "", {git, "git@git.herokai.com:heroku_crashdumps.git", {tag, "0.1.0"}}}
- ,{lager, "", {git, "git@git.herokai.com:lager.git", "2.0.0"}}
- ,{batchio, "", {git, "git@git.herokai.com:batchio.git", "1.0.0"}}
- ,{recon, "", {git, "git@git.herokai.com:recon.git", "master"}}
- ,{folsom, "", {git, "git@git.herokai.com:folsom.git", "0.8.2"}}
- ,{folsom_cowboy, "", {git, "git@git.herokai.com:folsom_cowboy.git", "51842c8e8a89707bdf072287703c721939d55827"}}
+  [{redis, {git, "git@git.herokai.com:erlang_redis_pool.git", {branch, "master"}}}
+  ,{mochiweb, {git, "git@git.herokai.com:mochiweb.git", {tag, "R16B01"}}}
+  ,{pagerduty, {git, "git@git.herokai.com:erlang_pagerduty.git", {branch, "master"}}}
```
Comment (srid) [inline rebar.config]: Just a question: a lot of repo deps point to `master` branch. Is `git.herokai.com` supposed to be a stable copy of these deps (manually updated by someone), or just a mirror?

Comment (voidlock) [inline rebar.config]: An internal mirror. To ensure deployments aren't blocked by a github outage.

Comment (srid) [inline rebar.config]: Ok, at least these repos are pinned in `rebar.lock`.

**Pull request merged**